scipy/optimize/optimize.py

```diff
@@ -421,37 +432,48 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
     retall = return_all
 
     fcalls, func = wrap_function(func, args)
-    x0 = asfarray(x0).flatten()
-    N = len(x0)
-    if maxiter is None:
-        maxiter = N * 200
-    if maxfun is None:
-        maxfun = N * 200
 
     rho = 1
     chi = 2
     psi = 0.5
     sigma = 0.5
-    one2np1 = list(range(1, N + 1))
+    nonzdelt = 0.05
+    zdelt = 0.00025
+
+    if initial_simplex is None:
+        x0 = asfarray(x0).flatten()
```

Comment:
You could move this outside the `if` clause, then you wouldn't need to do a very similar cast in the `else` clause below.